lib/config/configuration.go

```diff
@@ -661,26 +662,40 @@ func applyAuthConfig(fc *FileConfig, cfg *service.Config) error {
 		return trace.Wrap(err)
 	}
 
-	// Set cluster networking configuration from file configuration.
-	cfg.Auth.NetworkingConfig, err = types.NewClusterNetworkingConfigFromConfigFile(types.ClusterNetworkingConfigSpecV2{
-		ClientIdleTimeout:        fc.Auth.ClientIdleTimeout,
-		ClientIdleTimeoutMessage: fc.Auth.ClientIdleTimeoutMessage,
-		WebIdleTimeout:           fc.Auth.WebIdleTimeout,
-		KeepAliveInterval:        fc.Auth.KeepAliveInterval,
-		KeepAliveCountMax:        fc.Auth.KeepAliveCountMax,
-		SessionControlTimeout:    fc.Auth.SessionControlTimeout,
-		ProxyListenerMode:        fc.Auth.ProxyListenerMode,
-		RoutingStrategy:          fc.Auth.RoutingStrategy,
-		TunnelStrategy:           fc.Auth.TunnelStrategy,
-		ProxyPingInterval:        fc.Auth.ProxyPingInterval,
-	})
-	if err != nil {
-		return trace.Wrap(err)
+	// Only override networking configuration if some of its fields is
+	// specified in file configuration.
+	customNetworkingConfig := fc.Auth.ClientIdleTimeout != 0 ||
+		fc.Auth.ClientIdleTimeoutMessage != "" ||
+		fc.Auth.WebIdleTimeout != 0 ||
+		fc.Auth.KeepAliveInterval != 0 ||
+		fc.Auth.KeepAliveCountMax != 0 ||
+		fc.Auth.SessionControlTimeout != 0 ||
+		fc.Auth.ProxyListenerMode != 0 ||
+		fc.Auth.RoutingStrategy != 0 ||
+		fc.Auth.TunnelStrategy != nil ||
+		fc.Auth.ProxyPingInterval != 0
```

Comment:
:nail_care: 
Can we extract this check into a function?
And compare each field of the `fc.Auth` against a default `config.Auth` (as in `x := config.Auth{}`)?

Reply:
Thanks for the suggestion! Done in 857fd1e0ea06d03ba157d2b627f1e758600df414.